kai/incident_store.py

```diff
@@ -229,57 +234,195 @@ def find_solutions(
         pass
 
 
-# @dataclass
-# class Application:
-#     application_id: int | None
-#     application_name: str
-#     repo_uri_origin: str
-#     repo_uri_local: str
-#     current_branch: str
-#     current_commit: str
-#     generated_at: datetime.datetime
-
-#     def as_tuple(self):
-#         return (
-#             self.application_id,
-#             self.application_name,
-#             self.repo_uri_origin,
-#             self.repo_uri_local,
-#             self.current_branch,
-#             self.current_commit,
-#             self.generated_at,
-#         )
-
-#     @staticmethod
-#     def from_dict_row(row: DictRow) -> "Application":
-#         return Application(
-#             application_id=row["application_id"],
-#             application_name=row["application_name"],
-#             repo_uri_origin=row["repo_uri_origin"],
-#             repo_uri_local=row["repo_uri_local"],
-#             current_branch=row["current_branch"],
-#             current_commit=row["current_commit"],
-#             generated_at=row["generated_at"],
-#         )
-
-# @dataclass
-# class Ruleset:
-#     ruleset_name: str
-#     tags: dict
-
-# @dataclass
-# class Violation:
-#     violation_name: str
-#     category: str
-#     labels: dict
-
-# @dataclass
-# class Incident:
-#     incident_uri: str
-#     incident_snip: str
-#     incident_line: int
-#     incident_variables: dict
-#     application: Application
+# These classes are just for the in-memory data store. Once we figure out the
+# best way to model the incidents, violations, etc... we can remove these
+@dataclass(frozen=True)
```

Comment:
I'd create an incident_store directory and move the various implementations to their own files. Just makes it a little easier to read

Reply:
Done!